src/Runner.Worker/ActionManager.cs

```diff
@@ -1303,4 +1473,18 @@ public class ActionContainer
         public string WorkingDirectory { get; set; }
         public string ActionRepository { get; set; }
     }
+
+    public class ActionSetupInfo
+    {
+        public ActionContainer Container { get; set; }
+        public List<Pipelines.ActionStep> Steps {get; set;}
+    }
+
+    public class PrepareActionsState
+    {
+        public Dictionary<string, List<Guid>> ImagesToPull;
+        public Dictionary<string, List<Guid>> ImagesToBuild;
+        public Dictionary<string, ActionContainer> ImagesToBuildInfo;
```

Comment:
Consider combining `ImagesToBuild` and `ImagesToBuildInfo`

Reply:
Going to avoid refactoring this code for now since there are other refactors in this code and I want to minimize risk